aea/helpers/search/models.py

```diff
@@ -221,27 +221,33 @@ def _check_consistency(self):
 
         # check that all values are defined in the data model
         all_attributes = [attribute.name for attribute in self.data_model.attributes]
-        if not all(key in all_attributes for key in self.values):
+        if not all(key in all_attributes for key in self.values.keys()):
             raise AttributeInconsistencyException(
                 "Have extra attribute not in data model."
             )
 
-        # check that each of the values are consistent with that specified in the data model
-        for attribute in self.data_model.attributes:
-            if type(self.values[attribute.name]) != attribute.type:
+        # check that each of the provided values are consistent with that specified in the data model
+        for key, value in self.values.items():
```

Comment:
previous implementation did not consider the case when attributes are not required